### PR TITLE
Update of the login design document

### PR DESCRIPTION
**Pull request by pingchenchan**
## Description
I updated the login design doc based on [maarten/96-login-design](https://github.com/digitalaidseattle/plymouth-housing/tree/maarten/96-login-design). Add some needed features and questions, including the reset PIN flow for further discussion.

## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [x] Documentation Update

## Related Tickets & Documents
[PIT-96](https://das-ph-inventory-tracker.atlassian.net/browse/PIT-96?atlOrigin=eyJpIjoiZjhiMmMxMmQyMDU5NGI4MWEyYzdlNzVlMWJiZGFjMzYiLCJwIjoiaiJ9)

### CODE REVIEW

File: docs/designs/login-design.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+### Volunteer Table
+| Column Name      | Data Type | Description                          |
+|------------------|-----------|--------------------------------------|
+| `id`             | INT       | Primary key, auto-increment.         |
+| `name`     | VARCHAR   | Volunteer’s name(unique).              |
+| `email`  | VARCHAR   | Volunteer’s email address(unique).  |
+| `PIN`            | CHAR(4)   | 4-digit PIN for volunteer access.    |
+| `last_signed_in` | DATETIME  | Timestamp of the last sign-in.       |
+ |`activate`	 |BOOL |	Indicates if the volunteer is active (true/false). |
```
Comment (bospoort) [inline docs/designs/login-design.md:162]:
```suggestion
 |`active`	 |BOOL |	Indicates if the volunteer is active (true/false). |
```

File: docs/designs/login-design.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+- Integrate Azure OAuth configurations for authentication.
+- Ensure database backups and failover strategies are in place.
+
+### 8.2 Maintenance
+- Regularly update and patch the OAuth integration for security.
+- Monitor the database for performance and maintain data integrity.
+
+## 9. Risk Analysis
+- **Risk**: Unauthorized access if PINs are guessed.
+  - **Mitigation**: Implement rate limiting, captchas, and account lockout mechanisms after multiple failed attempts.
```
Comment (bospoort) [inline docs/designs/login-design.md:148]:
```suggestion
  - **Mitigation**: Implement account lockout mechanisms after three failed attempts.
```

File: docs/designs/login-design.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+- Simple PIN-based authentication for volunteers.
+- Data integrity and security for volunteer information.
+- Mechanism for PIN recovery via email.
+
+
+### 3.2 Constraints
+- Must use Azure OAuth for authentication.
+- Admin and volunteer dashboards are outside the scope of this design.
+- The solution must comply with data protection and privacy regulations.
+- The process for resetting a volunteer's PIN has not been finalized. It could be managed by admins via the admin page or by the volunteers themselves through a new PIN reset page.
```
Comment (bospoort) [inline docs/designs/login-design.md:47]: Can we start with a "forgot PIN email" that just sends the PIN back to the volunteer?

File: docs/designs/login-design.md [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+- **Volunteer PIN Page**: Allows volunteers to sign in using a PIN.
+- **Email Notification**: Sends email notifications for user to reset the PIN.
+- **Database**: Stores volunteer details, including the PIN, activation status, and last sign-in date.
+
+### 4.2 Component Descriptions
+- **Sign-In Page**: Initial landing page for all users; redirects to Azure OAuth for authentication.
+- **Azure OAuth Service**: Handles secure authentication; determines user role based on the email domain.
+- **Volunteer Name Page**: Displays a dropdown of volunteers names.
+- **Volunteer PIN Page**: Displays a PIN field; checks entered data against the database.
+- **Email Notification via Resend API**:  Sends email notifications using the Resend API through Supabase Edge Functions. This feature is used for sending important notifications, such as PIN reset emails to volunteers. [Resend API Documentation](https://resend.com/docs/send-with-supabase-edge-functions)
```
Comment (bospoort) [inline docs/designs/login-design.md:65]: Is this pending a discussion where PH wants to host? If they host on-prem, do we want/need another solution?

File: docs/designs/login-design.md [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+
+### 5.1 Subsystems and Components
+#### Sign-In Page
+- **Purpose**: Redirect users to Azure OAuth for authentication.
+- **Input**: User clicks the Azure sign-in button.
+- **Output**: Redirect to Azure OAuth login page.
+- **Dependencies**: Azure OAuth Service.
+
+#### Volunteer Name Page
+- **Purpose**: Allows volunteers to choice their name using a dropdown selection.
```
Comment (bospoort) [inline docs/designs/login-design.md:78]:
```suggestion
- **Purpose**: Allows volunteers to choose their name using a dropdown selection.
```

File: docs/designs/login-design.md [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+- **Input**: Selected volunteer name.
+- **Output**: Redirect to the volunteer Volunteer PIN Page.
+- **Dependencies**: Database for volunteer details.
+- **Additional Features**:
+  - **“I don’t see my name, what do I do now”** link: When clicked, a dialog will pop up. Provides a phone number or email of a PH admin that can sort it out (either get them activated in the database, and/or let them into another account for urgent matters)
+  - **Active User Filtering**: Only volunteers marked as "active" in the database will appear in the dropdown selection. This ensures that only currently active volunteers can sign in.
```
Comment (bospoort) [inline docs/designs/login-design.md:84]: Would this flag turn FALSE after 90 days?

**Review by bospoort — CHANGES_REQUESTED** [review]
Looks really good! Super complete. Can you resolve a few questions?

File: docs/designs/login-design.md (hunk 6 above)
Comment (pingchenchan) [reply docs/designs/login-design.md:84]: It sounds like a good idea.

File: docs/designs/login-design.md (hunk 4 above)
Comment (pingchenchan) [reply docs/designs/login-design.md:65]: I agree with you. It is important to figure out what they want.

File: docs/designs/login-design.md (hunk 3 above)
Comment (pingchenchan) [reply docs/designs/login-design.md:47]: I will update the Forgot PIN flow to what we discussed today by displaying a dialog box, "Please call the PH administrator for your PIN."

**Review by bospoort — APPROVED** [review]

**Pull request merged**